py/ngff_zarr/v05/zarr_metadata.py

```
@@ -26,18 +31,36 @@ def to_version(self, version: str | NgffVersion) -> "Metadata":
     @classmethod
-    def from_version(cls, metadata: "Metadata") -> "Metadata":
+    def from_version(cls, metadata: Union["Metadata", "Metadata_v04", "Metadata_v06"]) -> "Metadata":
         """Convert metadata from specified NGFF version."""
         from ..v04.zarr_metadata import Metadata as Metadata_v04
-
+        from ..v06.zarr_metadata import Metadata as Metadata_v06
+        
         if isinstance(metadata, Metadata_v04):
             return cls._from_v04(metadata)
-        raise ValueError(f"Unsupported metadata type: {type(metadata)}")
+        elif isinstance(metadata, Metadata_v06):
```

Ruff (0.15.12)

[warning] Unnecessary `elif` after `return` statement

Remove unnecessary `elif`

(RET505)